java/code/src/com/redhat/rhn/taskomatic/task/AutoErrataTask.java

```diff
@@ -94,6 +106,33 @@ public void execute(JobExecutionContext context)
         }
     }
 
+    /**
+     * Return the list of systems that have the auto errata update function enabled
+     *
+     * @return list of systems with auto errata update enabled
+     */
+    protected List<Map<String, Long>> getAutoErrataSystems() {
+        SelectMode select = ModeFactory.getMode(TaskConstants.MODE_NAME,
+                TaskConstants.TASK_QUERY_AUTO_ERRATA_SYSTEMS);
+        @SuppressWarnings("unchecked")
+        List<Map<String, Long>> results = select.execute();
+        return results;
+    }
+
+    /**
+     * Return list of systems that are in maintenance mode
+     *
+     * @param systems systems with auto errata enabled
+     * @return list of systems in maintenance mode
+     */
+    protected List<Long> filterSystemsInMaintenanceMode(List<Map<String, Long>> systems) {
+        MaintenanceManager mm = new MaintenanceManager();
+        return systems.stream()
+                .filter(system -> mm.isSystemInMaintenanceMode(ServerFactory.lookupById(system.get("id"))))
+                .map(system -> system.get("id"))
+                .collect(Collectors.toList());
```

Comment:
I'm not sure how efficiently Hibernate will handle this, but maybe querying all IDs at once is a better choice.
See: `ServerFactory.lookupByIds`